python/semantic_kernel/connectors/memory/azure_ai_search/azure_ai_search_collection.py

```diff
@@ -212,3 +228,107 @@ async def does_collection_exist(self, **kwargs) -> bool:
     @override
     async def delete_collection(self, **kwargs) -> None:
         await self.search_index_client.delete_index(self.collection_name, **kwargs)
+
+    @override
+    async def _inner_search(
```

Comment:
Based on our offline discussion yesterday, I can now understand how `_inner_search` needs to be implemented to satisfy all 3 mixins. A further question is: Do we need to check if only one of `search_text`, `vectorizable_text`, and `vector` is set?

Reply:
No, this is the inner method, so should not be called directly and the public functions only set one of those, I'll check if the docstring is clear on this.